src/ssh-generator/ssh-proxy.c

```diff
@@ -71,6 +71,52 @@ static int process_unix(const char *path) {
         if (r < 0)
                 return log_error_errno(r, "Failed to connect to AF_UNIX socket %s: %m", path);
 
+
+        r = send_one_fd_iov(STDOUT_FILENO, fd, &IOVEC_NUL_BYTE, /* n_iovec= */ 1, /* flags= */ 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to send socket via STDOUT: %m");
+
+        log_debug("Successfully sent AF_UNIX socket via STDOUT.");
+        return 0;
+}
+
+static int process_vsock_indirect(const char *path, const char *port) {
+        int r;
+
+        assert(path);
+        assert(port);
+
+        /* We assume the path is absolute unless it starts with a dot (or is already explicitly absolute) */
+        _cleanup_free_ char *prefixed = NULL;
+        if (!STARTSWITH_SET(path, "/", "./")) {
+                prefixed = strjoin("/", path);
+                if (!prefixed)
+                        return log_oom();
+
+                path = prefixed;
+        }
+
+        _cleanup_close_ int fd = socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0);
+        if (fd < 0)
+                return log_error_errno(errno, "Failed to allocate AF_UNIX socket: %m");
+
+        r = connect_unix_path(fd, AT_FDCWD, path);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to AF_UNIX socket %s: %m", path);
+
+        // based on the protocol as defined here:
+        // https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/vsock.md
+        // https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md
+        _cleanup_free_ char *connect_cmd = NULL;
+        connect_cmd = strjoin("CONNECT ", port, "\n");
+        if (!connect_cmd)
+                return log_oom();
+
+        int connect_cmd_len = strlen(connect_cmd);
```

Comment:
strlen() returns size_t (i.e. typically 64bit). hence change int → size_t as type for the variable (so that half of the bits are not dropped by accident)